example/node_classification.py

```diff
@@ -0,0 +1,131 @@
+    return loss
+
+
+
+def test(data, mask):
+    model.eval()
+    out = model(data.x, data.edge_index)
+    pred = out.argmax(dim=1)  # Use the class with highest probability.
+    correct = pred[mask] == data.y[mask]  # Check against ground-truth labels.
+    acc = int(correct.sum()) / int(mask.sum())  # Derive ratio of correct predictions.
+    return acc
+
+def convert_data(nodes, edges):
+    if len(nodes) == 0:
+        return "Insert data in database first!"
+    x = np.zeros((len(nodes),len(nodes[0].properties.get("feature"))))
+    y = np.zeros((len(nodes)))
+    edge_index = np.zeros((2,len(edges)))
+    train_mask = np.zeros((len(nodes)))
+    val_mask = np.zeros((len(nodes)))
+    test_mask = np.zeros((len(nodes)))
+    
+    print("Structuring data:")
+    for i in tqdm(range(len(nodes))):
+        x[i] = np.add(x[i], np.array(nodes[i].properties.get("feature")))
+        y[i] += nodes[i].properties.get("class")
+        train_mask[i] += nodes[i].properties.get("train_mask")
+        val_mask[i] += nodes[i].properties.get("val_mask")
+        test_mask[i] += nodes[i].properties.get("test_mask")
+
+    for i in tqdm(range(len(edges))):
+        edge_index[0][i] = edges[i].properties.get("first")
+        edge_index[1][i] = edges[i].properties.get("second")
+    print("Finished converting data.")
+
+    # print("=============================")
+    # print(np.shape(x))
+    # print(np.shape(y))
+    # print(np.shape(edge_index))
+    # print(np.shape(train_mask))
+    # print(np.shape(val_mask))
+    # print(np.shape(test_mask))
+    # print("=============================")
+
+
+    x = torch.tensor(x, dtype=torch.float32)
+    y = torch.tensor(y, dtype=torch.int64)
+    edge_index = torch.tensor(edge_index, dtype=torch.int64) 
+    train_mask = torch.tensor(train_mask, dtype=torch.bool)
+    val_mask = torch.tensor(val_mask, dtype=torch.bool)
+    test_mask = torch.tensor(test_mask, dtype=torch.bool)
+
+    data = Data(x, y, edge_index, train_mask, val_mask, test_mask)
+    return data
+
+
+@mgp.read_proc
+def work(
+    ctx: mgp.ProcCtx
+) -> mgp.Record(no_nodes=int, no_edges=int):
+    
+    nodes = list(iter(ctx.graph.vertices))
+    edges = []
+    for vertex in ctx.graph.vertices:
+        for edge in vertex.out_edges:
+            edges.append(edge)
+    
+    data = convert_data(nodes, edges)
+    if data == "Insert data in database first!":
+        print(data)
+        return mgp.Record(len_nodes=len(nodes), len_edges=len(edges))
+    
+    global model 
+    model = GCN(hidden_channels=16)
+    global optimizer 
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)
+    global criterion 
+    criterion = torch.nn.CrossEntropyLoss()
+    
+    for epoch in range(1, 101):
+        loss = train(data)
+        print(f'Epoch: {epoch:03d}, Loss: {loss:.4f}')
+    
+    test_acc = test(data, data.test_mask)
+    print(f'Test Accuracy: {test_acc:.4f}')
```

Comment:
While it probably doesn’t matter in a toy example like this one, **always** split the train and test sets.

Reply:
For now, training, validation and test data are separated by train_mask, val_mask, test_mask. If i-th example is for training, it has train_mask[i] = 1 and val_mask[i] = 0, test_mask[i] = 0.